analytic_engine/src/table/version.rs

```diff
@@ -806,11 +819,16 @@ impl TableVersionMeta {
         self.flushed_sequence = cmp::max(self.flushed_sequence, edit.flushed_sequence);
 
         for add_file in edit.files_to_add {
-            self.max_file_id = cmp::max(self.max_file_id, add_file.file.id);
+            // aligned max file id.
+            let aligned_max_file_id =
+                ((add_file.file.id - 1) / DEFAULT_ALLOC_STEP + 1) * DEFAULT_ALLOC_STEP;
+            self.max_file_id = cmp::max(self.max_file_id, aligned_max_file_id);
```

Comment:
Do align after L830. And in most cases, the alignment looks better if put it as `(a + step - 1) / step * step`.